GP2-667: Add enviroment-appropriate Wagtail Transfer configuration

stevejalim opened the pull request:
This changeset adds a function, `get_wagtail_transfer_configuration`, that provides the appropriate Wagtail-Transfer config for the environment the app is running on.

Its current scope is limited to to Beta, Staging, Dev or local development. We will add Production when the time is right.

### Why are we doing this?
We don't want to have a single config that basically allows and deployed site to import from any other deployed site, because this opens the door to content from Dev being imported to Beta (or Staging and then on to Beta) in error.

Includes tests.

Before this is merged the env vars for the deployed instances will need updating to ensure they have the appropriate values set:

Beta needs to know credentials for the single site it can pull from:

    WAGTAILTRANSFER_BASE_URL_STAGING
    WAGTAILTRANSFER_SECRET_KEY_STAGING

Staging needs to know credentials for the single site it can pull from:

    WAGTAILTRANSFER_BASE_URL_BETA
    WAGTAILTRANSFER_SECRET_KEY_BETA

Dev needs to know credentials for the TWO sites it can pull from:

    WAGTAILTRANSFER_BASE_URL_STAGING
    WAGTAILTRANSFER_SECRET_KEY_STAGING
    WAGTAILTRANSFER_BASE_URL_BETA
    WAGTAILTRANSFER_SECRET_KEY_BETA

## STATUS

 - [X] Ticket exists in Jira  https://uktrade.atlassian.net/browse/TICKET_ID_HERE 
 - [X] Jira ticket has the correct status.
 - [X] [Changelog](CHANGELOG.md) entry added.

 - [x] New environment variables (see above) added to vault. **PLEASE CHECK THESE AS PART OF REVIEW**

 - [X] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

depsiatwal commented on config/utils.py:
```diff
@@ -0,0 +1,85 @@
+    be set as settings.WAGTAILTRANSFER_SOURCES, which dictates where content
+    can be imported FROM.
+
+    For now we need the configuration to permit the following:
```
But long for a code comment may I suggest read.me also we then have single consolidated source for instructions

stevejalim replied:
I'll move it to the doc specific for Wagtail Transfer, sure

depsiatwal reviewed (commented):
LGTM one small comment

stevejalim commented:
@depsiatwal Updated - can I grab an approval and then I'll merge after I've set up the env vars?

stevejalim commented:
@depsiatwal I've made the followign additions;
* moved wagtail-transfer into running under the `/admin/` path, for neatness and easier exclusion from the ga360 middleware, if need be
* added the env vars in Vault too - would you mind taking a quick look to sense check them, please?

stevejalim commented:
Merging 

cc @agiamas FYI there are a bunch of new `WAGTAILTRANSFER_*` env vars in Vault
